clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -82,6 +83,14 @@ void CIRGenModule::emitGlobalFunctionDefinition(clang::GlobalDecl gd,
   theModule.push_back(funcOp);
 }
 
+void CIRGenModule::emitGlobalVarDefinition(const clang::VarDecl *vd,
+                                           bool isTentative) {
+  mlir::Type type = getTypes().convertType(vd->getType());
+  auto varOp = builder.create<cir::GlobalOp>(
+      getLoc(vd->getSourceRange()), vd->getIdentifier()->getName(), type);
```

Comment:
`getIdentifier` could potentially result in a nullptr if this is a 'special' name (or with some other magic? I know it is possible for local variables).  So an assert/fallback for `getIdentifier==nullptr` would be appreciated here.

Reply:
Fixed.  The code checks `getIdentifier()`.  If it is null, an error is reported and the declaration is skipped.  This is temporary until the code handles non-identifier names.